src/App/Fossa/Config/Analyze.hs

```diff
@@ -8,6 +8,8 @@ module App.Fossa.Config.Analyze (
   BinaryDiscovery (..),
   ExperimentalAnalyzeConfig (..),
   ForceVendoredDependencyRescans (..),
+  ForceVendoredDependencyCLILicenseScan (..),
+  ForceVendoredDependencyArchiveUpload (..),
```

Comment:
Since only one state is valid, please consider parsing this as an enum instead (ref: [ReportOutputFormat](https://github.com/fossas/fossa-cli/blob/102a09e1919379868739872e4939aff13529bf95/src/App/Fossa/Config/Report.hs#L60)), which lets us avoid an error case in general and is cleaner code-wise.

That'd be most easily done with a slight change in the CLI flag (maybe `--force-vendored-dependency-mode license-scan|archive-upload`), which leaves more room for future modes as well (I know we don't have any plans, but the flexibility would cost us roughly nothing right now, I think, since these are new flags).

Reply:
Yeah, you're right. I switched to this and the code is much cleaner

Thank you!